### PR TITLE
fix: fix emotion for storybook

### PR DESCRIPTION
**Pull request by ozgurkececioglu**
[FX-2655]

### Description

- fixed storybook configuration for new emotion usages
- updated some of the MUI utilities and theme structures

### How to test

- You can check some of the pages in the storybook. Don't forget that there are still some broken styles. It may be related to nested selectors which should be updated due to emotion updates. I will take care of them in the tss-react migration.

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2655]: https://toptal-core.atlassian.net/browse/FX-2655?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -94,7 +96,9 @@ module.exports = {
       '@toptal/picasso-provider': path.resolve(
         __dirname,
         '../packages/picasso-provider/src'
-      )
+      ),
+      '@emotion/core': toPath('node_modules/@emotion/react'),
+      'emotion-theming': toPath('node_modules/@emotion/react')
```
Comment (OndrejTuma) [inline .storybook/main.js:101]: can you use `require.resolve` here?

File: packages/picasso-provider/src/CssBaseline/styles.ts [hunk 2]
```diff
@@ -3,30 +3,28 @@ import createStyles from '@mui/styles/createStyles'
 
 export default ({ palette }: Theme) =>
   createStyles({
-    '@global': {
```
Comment (OndrejTuma) [inline packages/picasso-provider/src/CssBaseline/styles.ts:6]: I see: https://mui.com/material-ui/guides/migration-v4/#cssbaseline 👍

**Review by OndrejTuma — APPROVED** [review]
Code looks good 👍 I have only one nit, its on your consideration

File: .storybook/main.js [hunk 3]
```diff
@@ -2,6 +2,8 @@ const webpack = require('webpack')
 const path = require('path')
 const { IgnoreNotFoundPlugin } = require('./plugins')
 
+const toPath = filePath => path.join(process.cwd(), filePath)
```
Comment (OndrejTuma) [inline .storybook/main.js:5]: This is now a leftover I believe 🙂

**Review by augustobmoura — APPROVED** [review]
Aside from what @OndrejTuma pointed out, LGTM! :+1:

File: .storybook/main.js [hunk 4]
```diff
@@ -94,6 +94,14 @@ module.exports = {
...
         __dirname,
         '../packages/picasso-provider/src'
+      ),
+      '@emotion/core': path.resolve(
+        __dirname,
+        '../node_modules/@emotion/react'
+      ),
+      'emotion-theming': path.resolve(
+        __dirname,
+        '../node_modules/@emotion/react'
```
Comment (denieler) [inline .storybook/main.js:104]: @ozgurkececioglu is it intentional that they both are resolved to the same package?

Comment (ozgurkececioglu) [reply .storybook/main.js:104]: I saw this workaround here: https://mui.com/material-ui/guides/migration-v4/#storybook-emotion-with-v5

**Pull request merged**